### ISSUE TITLE
Make Date dialog - Date Picker for Origin

### ISSUE THREAD

**Issue by dannyparsons** [africanmathsinitiative/R-Instat#4550]
Currently we just have a combo box with two items for specifying the `origin` parameter to `as.Date`. But there are examples (e.g. NETcdf files) where we need to choose our own origin date. Therefore I would like to replace the combo box with a DatePicker control. We now have our DatePicker user control so it should be simple to replace this in the dialog.

**Comment by rdstern** [comment]
Do we want to replace the current options completely?  Excel and Gregorian may still be useful.

Could we change the label **Specify Origin** to just **Origin**.  Then we have 3 options, namely Excel, Gregorian and Specify.  If you choose Specify, then it opens the date picker?

**Comment by dannyparsons** [comment]
Yes I'm fine with that. The date picker could even be visible when you select Excel or Gregorian (maybe disabled) so that you can see the date it will use. Then you choose specify to enable and edit yourself?

**Comment by rdstern** [comment]
Additionally, these are examples where the variable may be numeric.  There is only one common example that could be numeric and you don't want to treat it as such.  That is the YEARMODA format (yyyymmdd).  I suggest this is the only common combination, because a) it is common and b) the other orders, e.g. ddmmyyyy are likely to be character variables anyway, so they keep the leading zero.

So I suggest, in addition, for the YEARMODA format, we add an **as.character**, so it doesn't give an error and ask for the origin, which is not the solution.  I assume, if it is already a character, then that addition still doesn't matter.

**Comment by dannyparsons** [comment]
I would suggest we need to restrict the receiver to non numeric columns when format is specified anyway, because the format option only works with non numeric columns. 
It means you would have to convert your numeric to character to be able to import but it should be clear because the receiver will not show numeric columns. That also makes the change to the dialog much easier.

**Comment by rdstern** [comment]
I have no problem with that solution, except that you are likely to  have selected the variable, before choosing **Specify Format** on the radio buttons.   So the column is already in the receiver.
So it isn't just a question of not allowing numeric columns in the selector with that option.  Will it also clear the receiver in that case?  If so, then fine.

**Comment by rdstern** [comment]
If it would clear the receiver, and this solution makes you more comfortable, then let's go for that.

**Comment by rdstern** [comment]
I am not sure of the proposed action for the YEARMODA format now?  I have just got data for Riga and this has the same format and comes as numeric.  Works fine when I change to character type.

**Comment by dannyparsons** [comment]
The simple solution is to restrict the receiver to non-numeric column when origin is selected. @shadrackkibet can't remember if this has been started already? If important for the next version now.

**Comment by dannyparsons** [comment]
Sorry that should be when **format** is specifed, not origin.

**Comment by shadrackkibet** [comment]
I am on top of this. Will finish up.

**Comment by shadrackkibet** [comment]
I think this was done , so this can be closed.
